cumulusci/tasks/salesforce/metadata_etl/base.py

```diff
@@ -0,0 +1,262 @@
+            )
+            types += base.format(members=members, name=entity)
+
+        return types
+
+    def _get_package_xml_content(self, deploy):
+        return f"""<?xml version="1.0" encoding="UTF-8"?>
+<Package xmlns="http://soap.sforce.com/2006/04/metadata">
+{self._get_types_package_xml()}
+    <version>{self.api_version}</version>
+</Package>
+"""
+
+    def _transform(self):
+        pass
+
+
+class MetadataSingleEntityTransformTask(BaseMetadataTransformTask):
+    """Base class for a Metadata ETL task that affects one or more
+    instances of a specific metadata entity."""
+
+    entity = None
+
+    task_options = {
+        "api_names": {"description": "List of API names of entities to affect"},
+        **BaseMetadataETLTask.task_options,
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+
+        self.api_names = list(
+            map(
+                self._namespace_injector,
+                process_list_arg(self.options.get("api_names", [])),
+            )
+        )
+
+    def _get_entities(self):
+        return {self.entity: self.api_names or ["*"]}
+
+    def _transform_entity(self, metadata, api_name):
+        return metadata
+
+    def _transform(self):
+        # call _transform_entity once per retrieved entity
+        # if the entity is an XML file, provide a parsed version
+        # and write the returned metadata into the deploy directory
+
+        parser = PackageXmlGenerator(
+            None, self.api_version
+        )  # We'll use it for its metadata_map
+        entity_configurations = list(
+            filter(
+                lambda entry: any(
+                    [
+                        subentry["type"] == self.entity
+                        for subentry in parser.metadata_map[entry]
+                    ]
+                ),
+                parser.metadata_map,
+            )
+        )
+        if not entity_configurations:
+            raise CumulusCIException(
+                f"Unable to locate configuration for entity {self.entity}"
+            )
+
+        configuration = parser.metadata_map[entity_configurations[0]][0]
+        if configuration["class"] not in [
+            "MetadataFilenameParser",
+            "CustomObjectParser",
+        ]:
+            raise CumulusCIException(
+                f"MetadataSingleEntityTransformTask only supports manipulating complete, file-based XML entities (not {self.entity})"
+            )
+
+        extension = configuration["extension"]
+        directory = entity_configurations[0]
+
+        for api_name in self.api_names:
+            path = self.retrieve_dir / directory / f"{api_name}.{extension}"
+            if not path.exists():
+                raise CumulusCIException(f"Cannot find metadata file {path}")
+
+            transformed_xml = self._transform_entity(
+                elementtree_parse_file(path), api_name
```

Comment:
Could we use lxml instead? @prescod has a PR #1391 which is trying to move us toward using it more consistently.

That said, I think we might want to figure out a way to do operations on the element tree without needing to specify the XML namespace everywhere.

Reply:
Proposal subsequent to discussion this morning: suppose instead of using `elementtree_parse_file()` this module explicitly uses `xml.etree` to guarantee to subclasses what they will receive, while preserving the insulation of the concrete tasks from disk access.

Then, when we move forward with switching to `lxml` across more domains (or wrapping it in an abstraction layer), I'll commit to refactoring the entire MD-ETL framework to use that.

Reply:
@davidmreed I'd rather proactively switch this over to lxml's etree, since that's very likely what we'll end up using as our standard internal representation of parsed xml. I realized just after our call that as long as we do that now, there's no particular urgency to come up with a convenience API that adapts the lxml etree.

I'll try to help with switching these tasks over to lxml in the next couple days.

Reply:
Done in https://github.com/SFDO-Tooling/CumulusCI/pull/1544

Reply:
Thank you, @davisagli!